SAP S/4HANA SOAP 2.0 Release (#691)
mulesoft/docs-connectors: modules/ROOT/pages/sap/sap-s4hana-soap-connector-studio.adoc

ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

Studio editors help you design and update your Mule applications, properties, and configuration files.

To configure a connector in Anypoint Studio:

. Create a Mule project.
. Supply your authentication credentials to Anypoint Platform.
. Add the connector to your Mule project.
. Configure the connector.

> Follow these best practices when you configure an input source and connector in Studio:

Reviewer: YAY LIST INTRO!

* Configure an associated global element for the connector.
* Create a YAML or properties file to contain your properties, and then reference the file from the configuration.

== Add the Connector to Your Project

> Add SAP S/4HANA SOAP Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as add the required dependencies to the project's pom.xml file.

Reviewer: ```suggestion
Add SAP S/4HANA SOAP Connector to a Mule project to automatically populate the XML code with the connector's namespace and schema location, as well as to add the required dependencies to the project's pom.xml file.
```

. In Studio, create a Mule project.

> . In the Mule Palette view, click *(X) Search in Exchange*.

Reviewer: ```suggestion
. In the *Mule Palette* view, click *(X) Search in Exchange*.
```

. In *Add Modules to Project*, type `SAP S/4HANA SOAP` in the search field.
. Click the SAP S/4HANA SOAP Cloud connector name in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

== Configure an Input Source

This connector does not provide an input source operation. You can configure an input source for the connector using one of the following: 

* *HTTP > Listener*
* *Core > Scheduler*

== Global Configuration Elements

To use SAP S/4HANA OData Cloud Connector in a Mule app, you can configure a global SAP S/4HANA configuration element for use by SAP S/4HANA connector operations in your application.

You can put placeholder values in the `mule-artifact.content` configuration file in the `src/main/resources` folder of your project.

You can either enter your credentials in the global configuration content or 

> reference a configuration file that contains the values for your credentials. 

Reviewer: Question: why did you split this sentence using + formatting, but not the longer sentences above it?

> Using a configuration file is recommended for simpler maintenance and better reuse of your project. Keeping these values in a separate file is useful if you 

Reviewer: ditto previous edit, same text, earlier
Also, change "may" to "might" in both instances.

need to deploy to different environments, such as production, development, and QA, where access credentials may differ.

=== Configure a Global Element

Use the Global Element Properties configuration dialog to configure a global element for SAP S/4HANA Cloud SOAP Connector:

. In your project, click the *Global Elements* tab at the base of the canvas.

> . In the *Global Configuration Elements* screen, click *Create*.

Reviewer: ```suggestion
. In the *Global Configuration Elements* screen, click *Create*:
```

+
image::sap/sap-hana-global-config-wizard.png[]
+
. In the *Choose Global Type* wizard, expand *Connector Configuration*, select *SAP S/4HANA Cloud SOAP Config*, and click *OK*.
. Configure the following values:
+
*General* tab:
. In the *Connection* drop-down, select the connection type:

>   * *Basic authentication* +

Reviewer: There is a formatting error here and I am not sure exactly what is going on.
The first-level step is configuring the following values, followed by a subhead of sorts, rather than a list item?
And then there is what looks like a substep that is not formatted as a substep and which has no peers.
Please examine and fix to transform as you intend; I will copyedit up to the point I think the formatting resumes validity.

Author: Good catch!

  For basic authentication, provide the following required values:
  ** Service Base URL +
  Enter the URL of the SAP S/4HANA instance you want to manage.
  ** Username +
  Enter the user name used to initialize the session.
  ** Password +
  Enter the password used to authenticate the user.
  * *OAuth 2 Application Connection Provider* +
  For OAuth 2.0 authentication, enter the following required information:
  ** Service base URL +
  Enter the base URL of the service provider of the SAP S/4HANA system to access from the connector. All other endpoints are derived from this base URL.
  ** Client ID +
  Enter the OAuth client ID that is registered with the service provider.
  ** Client secret +
  Enter the OAuth client secret that corresponds to the client ID and is registered with the service provider.

> . Enter the required information for the selected connection type, and click *OK*.

Reviewer: I think valid formatting restarts with this step.
The text above it looks like it should be bullets or substeps.

> . Click *Test Connection* to ensure your access credentials are correct.

Reviewer: ```suggestion
. Click *Test Connection* to ensure that your access credentials are correct.
```

. Click the *Advanced* tab to configure the following values:
+
[%header,cols="30s,70a"]
|===
|Studio Field |Description
|Discovery Service Enabled |Feature switch for service discovery. If disabled, the service discovery relative path is not applied.
|Discovery Service Path | Relative path from the base URL to the service to use for automatic services discovery.
|Service directory |Common HTTP directory for all SAP services. You can leave this parameter blank, and specify the HTTP directory individually for every service in the unlisted services list.
|Unlisted services |List of services to add to the list of discovered services.

> |Service discovery resolution timeout | Connection timeout for service discovery

Reviewer: ```suggestion
|Service discovery resolution timeout | Connection timeout for service discovery.
```

|Service discovery timeout unit | Time unit to use in the service discovery resolution timeout configuration.
|Default headers |Custom headers that can be included in each request sent.

> |Default query parameter |Custom query parameters to include in each request sent

Reviewer: ```suggestion
|Default query parameter |Custom query parameters to include in each request sent.
```

> |TLS configuration |TLS configuration

Reviewer: ```suggestion
|TLS configuration |TLS configuration.
```

> |Proxy configuration |Configuration for executing requests through a proxy

Reviewer: ```suggestion
|Proxy configuration |Configuration for executing requests through a proxy.
```

|===
+
. Click *OK* to save the global connector configuration.

== Configure Authentication

> To access the data in the SAP S/4HANA instance, you need to authenticate your application’s requests using a supported authentication method. SAP S/4HANA Cloud SOAP Connector supports the following authentication types:

Reviewer: ```suggestion
To access the data in the SAP S/4HANA instance, authenticate your application’s requests using a supported authentication method. SAP S/4HANA Cloud SOAP Connector supports the following authentication types:
```

Reviewer: Actually, this text would be simpler by including the two types in it, since there are only two:
```suggestion
To access the data in the SAP S/4HANA instance, your application’s requests using either Basic or OAuth 2.0.
```

* Basic 
* OAuth 2.0

> 

Reviewer: See previous comment.

> To configuration the connection:

Reviewer: ```suggestion
To configure the connection:
```

. Drag an SAP S/4HANA Cloud connector operation to the canvas.
. Click the operation.

> . Click the green plus sign on the right of *Connector Configuration*.

Reviewer: ```suggestion
. Click the green plus sign (+) to the right of *Connector Configuration*.
```

. In the *Connection* drop-down, select the connection type:
  * *Basic authentication* +
  For basic authentication, provide the following required values:
  ** Service Base URL +
  Enter the URL of the SAP S/4HANA instance you want to manage.
  ** Username +
  Enter the user name used to initialize the session.
  ** Password +
  Enter the password used to authenticate the user.
  * *OAuth 2 Application Connection Provider* +
  For OAuth 2.0 authentication, enter the following required information:
  ** Service base URL +
  Enter the base URL of the service provider of the SAP S/4HANA system to access from the connector. All other endpoints are derived from this base URL.
  ** Client ID +
  Enter the OAuth client ID that is registered with the service provider.
  ** Client secret +
  Enter the OAuth client secret that corresponds to the client ID and is registered with the service provider.

> . Enter the required information for the selected connection type, and click *OK*.

Reviewer: ```suggestion
. Enter the required information for the selected connection type and click *OK*.
```
